### DISCUSSION TITLE
[v2] How to access the hydrated model instance for package developers

### DISCUSSION THREAD

**Discussion by tanthammar** [livewire/livewire#1477] [4. General]
As a package developer I need to know if I can access the hydrated model instance?

My problem is that I want users of the package to be able to use the new model binding like:
```php

User $user;

//no mount method
```

But I am unable to figure out a way to work with the hydrated model instance. 

I would like a new protected property - example name`$hydratedModel` that I can use everywhere like `$this->hydratedModel`

I tried this:

```php

User $user;

public function mount()
{
    $this->my_package_mount('user');
}
```

In my package
```php
public $hydratedModel;

public function my_package_mount($modelName)
{
    $this->hydratedModel = $this->$modelName;
}
```

But I get the error that `$user must not be accessed before initialization`

**Comment by tanthammar** [comment]
To setup everything I need the model in the mount method.

**Comment by tanthammar** [comment]
@nuernbergerA, @calebporzio  Do you have any ideas?

**Comment by inxilpro** [comment]
By the time `mount()` is called, the instance of the model should be initialized. Can you share a full example or point to a repository of a package that's not working?

**Comment by tanthammar** [comment]
Just add this to a component and you'll get the error

```php
    public User $user;
    public $hydratedModel;

public function mount()
{
    $this->hydratedModel = $this->user;
}
```

**Comment by tanthammar** [comment]
Got it down to Livewire routes with optional parameters #1547

**Comment by inxilpro** [comment]
Ah! I see — yeah, now if you want an optional parameter to be initialized with a new, unsaved instance, you need to do that in your `mount()` method. I guess we could change that, but I feel like it's kind of counter-intuitive. 

My expectation would be that if a parameter is optional in the route, I would make it nullable in my component and then check for null in the view or mount up a default in `mount()`.

**Reply by tanthammar** [reply]
Can you please comment or - if this is the final/official Livewire reply - close issue #1547 
Just want to know if this will or will not, be implemented, so I can document and continue developing my package.

**Reply by inxilpro** [reply]
I think @calebporzio will need to chime in here. My preference would be to leave it as is, but I get that it's confusing because it's slightly different behavior than v1.
